Shaders/VXGI.shader

```diff
@@ -49,6 +53,37 @@ Shader "Hidden/VXGI"
         return o;
       }
 
+      static int3 GatherOffsets[4] = {
+        int3(0, 1, 0),
+        int3(1, 1, 0),
+        int3(1, 0, 0),
+        int3(0, 0, 0)
+      };
+
+      int3 NearestDepthTexel(float2 uv, Texture2D sourceDepths, float4 sourceTexelSize, float targetDepth, float4 targetTexelSize)
+      {
+        int3 texel = int3(mad(uv, sourceTexelSize.zw, -0.5), 0);
+
+        if (all(sourceTexelSize.zw == targetTexelSize.zw)) return texel;
+
+        float4 neighborDepths = sourceDepths.Gather(point_clamp_sampler, uv);
+
+        int minIndex = 0;
+        float minDist = distance(targetDepth, neighborDepths[0]);
+
+        [unroll]
+        for (int i = 1; i < 4; i++) {
+          float dist = distance(targetDepth, neighborDepths[i]);
+
+          if (dist >= minDist) continue;
+
+          minDist = dist;
+          minIndex = i;
+        }
+
+        return texel + GatherOffsets[minIndex];
```

Comment:
@jeske I think there is a problem, the `minIndex` here seems to stay zero all the time. Can you help me check this out?

![Capture](https://user-images.githubusercontent.com/8206622/59054074-17bd7700-88bd-11e9-8ad0-023d9db46430.PNG)

Reply:
## 1/8 resolution
![1/8 resolution](https://user-images.githubusercontent.com/8206622/59055684-9bc52e00-88c0-11e9-9e6b-d44121d3327f.PNG)

---

## 1/4 resolution
![1/4 resolution](https://user-images.githubusercontent.com/8206622/59055763-d4fd9e00-88c0-11e9-95f3-86175c4a8488.PNG)